src/sonic-frr/patch/0037-bgp-community-memory-leak-fix.patch

```diff
@@ -0,0 +1,470 @@
+ 						bgp_attr_intern(&attr);
+ 
+-					bgp_adj_out_set_subgroup(dest, subgrp,
+-								 adv_attr,
+-								 selected);
+-				} else
++					if (!bgp_adj_out_set_subgroup(dest,
++								      subgrp,
++								      adv_attr,
++								      selected))
++						bgp_attr_flush(&attr);
++				} else {
+ 					bgp_adj_out_unset_subgroup(
+ 						dest, subgrp, 1, addpath_tx_id);
+-			}
+-		} else
++					bgp_attr_flush(&attr);
++				}
++			} else
++				bgp_attr_flush(&attr);
++		} else {
+ 			bgp_adj_out_unset_subgroup(dest, subgrp, 1,
+ 						   addpath_tx_id);
++			bgp_attr_flush(&attr);
++		}
+ 	}
+ 
+ 	/* If selected is NULL we must withdraw the path using addpath_tx_id */
+diff --git a/bgpd/bgp_updgrp.h b/bgpd/bgp_updgrp.h
+index e27c1e7b67..b7b6aa07e9 100644
+--- a/bgpd/bgp_updgrp.h
++++ b/bgpd/bgp_updgrp.h
+@@ -458,7 +458,7 @@ extern struct bgp_adj_out *bgp_adj_out_alloc(struct update_subgroup *subgrp,
+ extern void bgp_adj_out_remove_subgroup(struct bgp_dest *dest,
+ 					struct bgp_adj_out *adj,
+ 					struct update_subgroup *subgrp);
+-extern void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
++extern bool bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 				     struct update_subgroup *subgrp,
+ 				     struct attr *attr,
+ 				     struct bgp_path_info *path);
+diff --git a/bgpd/bgp_updgrp_adv.c b/bgpd/bgp_updgrp_adv.c
+index de2b3206b7..d566fc15ad 100644
+--- a/bgpd/bgp_updgrp_adv.c
++++ b/bgpd/bgp_updgrp_adv.c
+@@ -454,7 +454,7 @@ bgp_advertise_clean_subgroup(struct update_subgroup *subgrp,
+ 	return next;
+ }
+ 
+-void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
++bool bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 			      struct update_subgroup *subgrp, struct attr *attr,
+ 			      struct bgp_path_info *path)
+ {
+@@ -474,7 +474,7 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 	bgp = SUBGRP_INST(subgrp);
+ 
+ 	if (DISABLE_BGP_ANNOUNCE)
+-		return;
++		return false;
+ 
+ 	/* Look for adjacency information. */
+ 	adj = adj_lookup(
+@@ -490,7 +490,7 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 			bgp_addpath_id_for_peer(peer, afi, safi,
+ 						&path->tx_addpath));
+ 		if (!adj)
+-			return;
++			return false;
+ 
+ 		subgrp->pscount++;
+ 	}
+@@ -529,7 +529,7 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 		 * will never be able to coalesce the 3rd peer down
+ 		 */
+ 		subgrp->version = MAX(subgrp->version, dest->version);
+-		return;
++		return false;
+ 	}
+ 
+ 	if (adj->adv)
+@@ -576,6 +576,8 @@ void bgp_adj_out_set_subgroup(struct bgp_dest *dest,
+ 	bgp_adv_fifo_add_tail(&subgrp->sync->update, adv);
+ 
+ 	subgrp->version = MAX(subgrp->version, dest->version);
++
++	return true;
+ }
+ 
+ /* The only time 'withdraw' will be false is if we are sending
+@@ -719,9 +721,10 @@ void subgroup_announce_table(struct update_subgroup *subgrp,
+ 						struct attr *adv_attr =
+ 							bgp_attr_intern(&attr);
+ 
+-						bgp_adj_out_set_subgroup(
++						if (!bgp_adj_out_set_subgroup(
```

Comment:
@dgsudharsan  as we discussed, 
1. No need to check the return value of this function, either true  or false, we need to free. 
2. In the line 376, to replace bgp_attr_flush with bgp_attr_unintern, which is coupled with bgp_attr_intern.
3. In the else branch, to add the  bgp_attr_flush.

Reply:
Seems the latest patch still checking the function and no flush in the else patch, Could you please help to double confirm, thanks a lot.